src/file.cpp

```diff
@@ -336,11 +336,11 @@ static_assert((libtorrent::file::sparse & libtorrent::file::attribute_mask) == 0
 
 namespace libtorrent
 {
-	int bufs_size(file::iovec_t const* bufs, int num_bufs)
+	int bufs_size(span<file::iovec_t const> bufs)
 	{
 		std::size_t size = 0;
-		for (file::iovec_t const* i = bufs, *end(bufs + num_bufs); i < end; ++i)
-			size += i->iov_len;
+		for (auto buf : bufs)
```

Comment:
If the objects you're dealing with are large then yes, you would want to use auto const&. I was speaking specifically about spans which are small and trivial to copy so passing by value is preferable.